app/models/document.rb

```diff
@@ -91,19 +89,11 @@ def metadata_value_present?(metadata_hash)
 
   def humanize_metadata_name(metadata_hash)
     metadata_hash.merge(
-      name: metadata_label(metadata_hash.fetch(:name))
+      name: metadata_name_mappings.fetch(metadata_hash.fetch(:name))
     )
   end
 
-  def metadata_label(key)
-    self.class.metadata_name_mappings.fetch(key, key.humanize)
-  end
-
-  def truncated_summary_or_nil
-    description = attrs.fetch(:description, nil)
-
-    # This truncates the description at the end of the first sentence
-    description = description.gsub(/\.\s[A-Z].*/, '.') if description.present?
-    description
+  def metadata_name_mappings
+    finder.metadata.each_with_object({}) { |facet, hash| hash[facet.key] = finder.label_for_metadata_key(facet.key) }
```

Comment:
What does producing this lookup table give us that we can't get by just calling the finder directly? Ie:

```ruby
  def humanize_metadata_name(metadata_hash)
    metadata_hash.merge(
      name: finder.label_for_metadata_key(metadata_hash.fetch(:name))
    )
  end
```